File: EBiletJetpack/app/src/main/java/com/alperensarac/ebiletjetpack/ui/event/EventDetailScreen.kt
package com.alperensarac.ebiletjetpack.ui.event


import android.widget.Toast
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import coil.compose.AsyncImage
import com.alperensarac.ebiletjetpack.data.api.ApiClient
import com.alperensarac.ebiletjetpack.data.model.ApiResponse
import com.alperensarac.ebiletjetpack.data.model.Event
import com.alperensarac.ebiletjetpack.data.model.Ticket
import com.alperensarac.ebiletjetpack.data.session.SessionManager
import com.alperensarac.ebiletjetpack.ui.components.AppBackground
import com.alperensarac.ebiletjetpack.ui.components.AppBlue
import com.alperensarac.ebiletjetpack.ui.components.AppGreen
import com.alperensarac.ebiletjetpack.ui.components.AppSmallWhiteButton
import com.alperensarac.ebiletjetpack.ui.components.AppTextDark
import com.alperensarac.ebiletjetpack.ui.components.AppTextGray
import com.alperensarac.ebiletjetpack.ui.components.AppTopBar
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

/*
    EventDetailScreen

    Etkinlik detay ekranı.

    HomeScreen'den buraya sadece eventId gelir.

    Neden sadece eventId?
    Çünkü detay ekranında en güncel veriyi backend'den almak daha doğru.
    Örneğin:
    - Kontenjan değişmiş olabilir.
    - Etkinlik pasife alınmış olabilir.
    - Fiyat güncellenmiş olabilir.

    Kullanılan API'ler:

    1. events/event_detail.php
       Etkinlik detayını getirir.

    2. tickets/ticket_buy.php
       Kullanıcı için bilet oluşturur.
*/
@Composable
fun EventDetailScreen(
    eventId: Int,
    onBack: () -> Unit,
    onTicketBought: () -> Unit
) {
    val context = LocalContext.current

    val sessionManager = remember {
        SessionManager(context)
    }

    /*
        Backend'den gelen etkinlik.
    */
    var event by remember { mutableStateOf<Event?>(null) }

    /*
        Genel durum mesajı.
    */
    var statusMessage by remember { mutableStateOf("Etkinlik detayı yükleniyor...") }

    /*
        Detay yükleniyor mu?
    */
    var isLoading by remember { mutableStateOf(false) }

    /*
        Bilet satın alma isteği devam ediyor mu?
    */
    var isBuying by remember { mutableStateOf(false) }

    /*
        Bu fonksiyon etkinlik detayını backend'den çeker.

        Compose içinde aynı kodu hem ilk açılışta hem de bilet aldıktan sonra
        tekrar kullanmak isteyebiliriz. Bu yüzden local fonksiyon olarak yazıyoruz.
    */
    fun loadEventDetail() {
        isLoading = true
        statusMessage = "Etkinlik detayı yükleniyor..."

        ApiClient.apiService
            .getEventDetail(
                apiToken = sessionManager.getApiToken(),
                eventId = eventId
            )
            .enqueue(object : Callback<ApiResponse<Event>> {

                override fun onResponse(
                    call: Call<ApiResponse<Event>>,
                    response: Response<ApiResponse<Event>>
                ) {
                    isLoading = false

                    if (!response.isSuccessful) {
                        statusMessage = "Sunucu hatası: ${response.code()}"
                        return
                    }

                    val apiResponse = response.body()

                    if (apiResponse == null) {
                        statusMessage = "Boş sunucu cevabı"
                        return
                    }

                    if (!apiResponse.success) {
                        statusMessage = apiResponse.message
                        return
                    }

                    val loadedEvent = apiResponse.data

                    if (loadedEvent == null) {
                        statusMessage = "Etkinlik bilgisi alınamadı"
                        return
                    }

                    event = loadedEvent
                    statusMessage = "Etkinlik detayı getirildi."
                }

                override fun onFailure(
                    call: Call<ApiResponse<Event>>,
                    t: Throwable
                ) {
                    isLoading = false
                    statusMessage = "Bağlantı hatası: ${t.localizedMessage}"
                }
            })
    }

    /*
        Ekran ilk açıldığında detay çekilir.
    */
    LaunchedEffect(eventId) {
        if (eventId <= 0) {
            statusMessage = "Etkinlik bilgisi alınamadı"
            return@LaunchedEffect
        }

        loadEventDetail()
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(AppBackground)
    ) {
        AppTopBar(
            title = "Etkinlik Detayı",
            subtitle = event?.title,
            actions = {
                AppSmallWhiteButton(
                    text = "Geri",
                    onClick = onBack
                )
            }
        )

        Column(
            modifier = Modifier
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
                .padding(14.dp),
            verticalArrangement = Arrangement.spacedBy(14.dp)
        ) {
            Text(
                text = statusMessage,
                color = AppTextGray
            )

            if (isLoading && event == null) {
                LoadingCard()
            }

            val currentEvent = event

            if (currentEvent != null) {
                EventDetailContent(
                    event = currentEvent,
                    isBuying = isBuying,
                    onBuyTicket = {
                        val remainingQuota = currentEvent.remainingQuota ?: 0

                        if (remainingQuota <= 0) {
                            Toast.makeText(
                                context,
                                "Bu etkinlik için kontenjan kalmamış",
                                Toast.LENGTH_SHORT
                            ).show()
                            return@EventDetailContent
                        }

                        isBuying = true
                        statusMessage = "Bilet oluşturuluyor..."

                        ApiClient.apiService
                            .buyTicket(
                                apiToken = sessionManager.getApiToken(),
                                eventId = currentEvent.id
                            )
                            .enqueue(object : Callback<ApiResponse<Ticket>> {

                                override fun onResponse(
                                    call: Call<ApiResponse<Ticket>>,
                                    response: Response<ApiResponse<Ticket>>
                                ) {
                                    isBuying = false

                                    if (!response.isSuccessful) {
                                        statusMessage = "Sunucu hatası: ${response.code()}"
                                        return
                                    }

                                    val apiResponse = response.body()

                                    if (apiResponse == null) {
                                        statusMessage = "Boş sunucu cevabı"
                                        return
                                    }

                                    if (!apiResponse.success) {
                                        statusMessage = apiResponse.message

                                        Toast.makeText(
                                            context,
                                            apiResponse.message,
                                            Toast.LENGTH_LONG
                                        ).show()

                                        return
                                    }

                                    val ticket = apiResponse.data
                                    val ticketCode = ticket?.ticketCode ?: "-"

                                    Toast.makeText(
                                        context,
                                        "Bilet alındı: $ticketCode",
                                        Toast.LENGTH_LONG
                                    ).show()

                                    statusMessage = "Bilet başarıyla oluşturuldu."

                                    /*
                                        Kullanıcı biletini QR ile görebilsin diye
                                        Biletlerim ekranına yönlendiriyoruz.
                                    */
                                    onTicketBought()
                                }

                                override fun onFailure(
                                    call: Call<ApiResponse<Ticket>>,
                                    t: Throwable
                                ) {
                                    isBuying = false
                                    statusMessage = "Bağlantı hatası: ${t.localizedMessage}"
                                }
                            })
                    }
                )
            }
        }
    }
}

/*
    Detay yüklenirken gösterilen basit kart.
*/
@Composable
private fun LoadingCard() {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(18.dp),
        colors = CardDefaults.cardColors(
            containerColor = Color.White
        )
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .padding(24.dp),
            contentAlignment = Alignment.Center
        ) {
            Text(
                text = "Yükleniyor...",
                color = AppTextGray
            )
        }
    }
}

/*
    Etkinlik detay içeriği.

    Bu composable sadece UI basar.
    API isteği yapmaz.

    Böyle ayırmamızın sebebi:
    - Ana screen state ve API yönetir.
    - Content sadece ekrana veri basar.
    - Kod okunabilir olur.
*/
@Composable
private fun EventDetailContent(
    event: Event,
    isBuying: Boolean,
    onBuyTicket: () -> Unit
) {
    val posterUrl = event.posterUrl?.let { poster ->
        if (poster.startsWith("http")) {
            poster
        } else {
            ApiClient.BASE_URL + poster
        }
    }

    AsyncImage(
        model = posterUrl,
        contentDescription = event.title,
        modifier = Modifier
            .fillMaxWidth()
            .height(240.dp)
            .background(
                color = Color(0xFFE2E8F0),
                shape = RoundedCornerShape(18.dp)
            )
    )

    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(18.dp),
        colors = CardDefaults.cardColors(
            containerColor = Color.White
        ),
        elevation = CardDefaults.cardElevation(
            defaultElevation = 4.dp
        )
    ) {
        Column(
            modifier = Modifier.padding(16.dp)
        ) {
            Text(
                text = event.title,
                style = MaterialTheme.typography.headlineSmall,
                fontWeight = FontWeight.Bold,
                color = AppTextDark
            )

            Spacer(modifier = Modifier.height(10.dp))

            Text(
                text = event.description ?: "Açıklama bulunmuyor.",
                color = Color(0xFF475569)
            )

            Spacer(modifier = Modifier.height(18.dp))

            DetailLine(
                title = "Tarih",
                value = event.eventDate ?: "-"
            )

            val cityName = event.city?.name ?: event.cityName ?: "-"
            val districtName = event.district?.name ?: event.districtName ?: "-"

            DetailLine(
                title = "Konum",
                value = "$cityName / $districtName"
            )

            DetailLine(
                title = "Sahne",
                value = event.venue?.name ?: "-"
            )

            DetailLine(
                title = "Adres",
                value = event.venue?.address ?: "-"
            )

            Spacer(modifier = Modifier.height(18.dp))

            Row(
                modifier = Modifier.fillMaxWidth(),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "${event.basePrice?.toInt() ?: 0} TL",
                    color = AppGreen,
                    style = MaterialTheme.typography.headlineSmall,
                    fontWeight = FontWeight.Bold,
                    modifier = Modifier.weight(1f)
                )

                Box(
                    modifier = Modifier
                        .background(
                            color = Color(0xFFEFF6FF),
                            shape = RoundedCornerShape(12.dp)
                        )
                        .padding(
                            horizontal = 12.dp,
                            vertical = 8.dp
                        )
                ) {
                    Text(
                        text = "Kalan: ${event.remainingQuota ?: 0}",
                        color = AppBlue,
                        fontWeight = FontWeight.Bold
                    )
                }
            }

            Spacer(modifier = Modifier.height(22.dp))

            val remainingQuota = event.remainingQuota ?: 0
            val canBuy = remainingQuota > 0 && !isBuying

            Button(
                onClick = onBuyTicket,
                enabled = canBuy,
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(14.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = if (remainingQuota > 0) {
                        AppGreen
                    } else {
                        Color(0xFF94A3B8)
                    },
                    contentColor = Color.White
                )
            ) {
                Text(
                    text = when {
                        isBuying -> "Bilet Oluşturuluyor..."
                        remainingQuota <= 0 -> "Kontenjan Doldu"
                        else -> "Bilet Al"
                    }
                )
            }
        }
    }
}

/*
    Detay satırı.

    Örnek:
    Tarih: 2026-07-10 20:30:00
*/
@Composable
private fun DetailLine(
    title: String,
    value: String
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 5.dp)
    ) {
        Text(
            text = title,
            color = AppTextGray,
            style = MaterialTheme.typography.bodySmall
        )

        Text(
            text = value,
            color = AppTextDark,
            fontWeight = FontWeight.Bold
        )
    }
}

File: ebilet_flutter/lib/screens/event/event_detail_screen.dart
import 'package:flutter/material.dart';

import '../../core/api_client.dart';
import '../../core/api_service.dart';
import '../../core/app_colors.dart';
import '../../core/session_manager.dart';
import '../../models/event_model.dart';
import '../../widgets/app_button.dart';

class EventDetailScreen extends StatefulWidget {
  final int eventId;

  const EventDetailScreen({
    super.key,
    required this.eventId,
  });

  @override
  State<EventDetailScreen> createState() => _EventDetailScreenState();
}

class _EventDetailScreenState extends State<EventDetailScreen> {
  EventModel? _event;

  bool _isLoading = false;
  bool _isBuying = false;

  String _statusMessage = 'Etkinlik detayı yükleniyor...';

  @override
  void initState() {
    super.initState();
    _loadEventDetail();
  }

  Future<void> _loadEventDetail() async {
    setState(() {
      _isLoading = true;
      _statusMessage = 'Etkinlik detayı yükleniyor...';
    });

    try {
      final apiToken = await SessionManager.getApiToken();

      final response = await ApiService.getEventDetail(
        apiToken: apiToken,
        eventId: widget.eventId,
      );

      if (!mounted) return;

      setState(() {
        _isLoading = false;
      });

      if (!response.success) {
        setState(() {
          _statusMessage = response.message;
        });
        return;
      }

      if (response.data == null) {
        setState(() {
          _statusMessage = 'Etkinlik bilgisi alınamadı.';
        });
        return;
      }

      setState(() {
        _event = response.data;
        _statusMessage = 'Etkinlik detayı getirildi.';
      });
    } catch (e) {
      if (!mounted) return;

      setState(() {
        _isLoading = false;
        _statusMessage = e.toString();
      });
    }
  }

  Future<void> _buyTicket() async {
    final event = _event;

    if (event == null) {
      _showMessage('Etkinlik bilgisi bulunamadı.');
      return;
    }

    final remainingQuota = event.remainingQuota ?? 0;

    if (remainingQuota <= 0) {
      _showMessage('Bu etkinlik için kontenjan kalmamış.');
      return;
    }

    setState(() {
      _isBuying = true;
      _statusMessage = 'Bilet oluşturuluyor...';
    });

    try {
      final apiToken = await SessionManager.getApiToken();

      final response = await ApiService.buyTicket(
        apiToken: apiToken,
        eventId: event.id,
      );

      if (!mounted) return;

      setState(() {
        _isBuying = false;
      });

      if (!response.success) {
        setState(() {
          _statusMessage = response.message;
        });

        _showMessage(response.message);
        return;
      }

      final ticketCode = response.data?.ticketCode ?? '-';

      setState(() {
        _statusMessage = 'Bilet başarıyla oluşturuldu.';
      });

      _showTicketSuccessDialog(ticketCode);
    } catch (e) {
      if (!mounted) return;

      setState(() {
        _isBuying = false;
        _statusMessage = e.toString();
      });

      _showMessage(e.toString());
    }
  }

  void _showTicketSuccessDialog(String ticketCode) {
    showDialog(
      context: context,
      builder: (dialogContext) {
        return AlertDialog(
          title: const Text('Bilet Alındı'),
          content: Text('Bilet kodu: $ticketCode'),
          actions: [
            TextButton(
              onPressed: () {
                Navigator.pop(dialogContext);

                /*
                  Bir sonraki adımda MyTicketsScreen yapınca
                  burada direkt Biletlerim ekranına yönlendireceğiz.
                */
                Navigator.pop(context);
              },
              child: const Text('Tamam'),
            ),
          ],
        );
      },
    );
  }

  void _showMessage(String message) {
    if (!mounted) return;

    ScaffoldMessenger.of(context).showSnackBar(
      SnackBar(
        content: Text(message),
      ),
    );
  }

  String _posterUrl(EventModel event) {
    final poster = event.posterUrl ?? '';

    if (poster.isEmpty) {
      return '';
    }

    if (poster.startsWith('http')) {
      return poster;
    }

    return ApiClient.baseUrl + poster;
  }

  bool _canBuy(EventModel event) {
    final remainingQuota = event.remainingQuota ?? 0;
    return remainingQuota > 0 && !_isBuying && !_isLoading;
  }

  String _buyButtonText(EventModel event) {
    if (_isBuying) {
      return 'Bilet Oluşturuluyor...';
    }

    final remainingQuota = event.remainingQuota ?? 0;

    if (remainingQuota <= 0) {
      return 'Kontenjan Doldu';
    }

    return 'Bilet Al';
  }

  Color _buyButtonColor(EventModel event) {
    final remainingQuota = event.remainingQuota ?? 0;

    if (remainingQuota <= 0) {
      return Colors.grey;
    }

    return AppColors.green;
  }

  @override
  Widget build(BuildContext context) {
    final event = _event;

    return Scaffold(
      backgroundColor: AppColors.background,
      appBar: AppBar(
        title: const Text('Etkinlik Detayı'),
        backgroundColor: AppColors.background,
        foregroundColor: AppColors.darkText,
        elevation: 0,
      ),
      body: SafeArea(
        child: RefreshIndicator(
          onRefresh: _loadEventDetail,
          child: ListView(
            padding: const EdgeInsets.all(14),
            children: [
              _buildStatusRow(),
              const SizedBox(height: 12),
              if (_isLoading && event == null) _buildLoadingCard(),
              if (!_isLoading && event == null) _buildEmptyCard(),
              if (event != null) _buildContent(event),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildStatusRow() {
    return Row(
      children: [
        if (_isLoading || _isBuying) ...[
          const SizedBox(
            width: 18,
            height: 18,
            child: CircularProgressIndicator(strokeWidth: 2),
          ),
          const SizedBox(width: 8),
        ],
        Expanded(
          child: Text(
            _statusMessage,
            style: const TextStyle(
              fontSize: 14,
              color: AppColors.grayText,
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildLoadingCard() {
    return Container(
      width: double.infinity,
      padding: const EdgeInsets.all(24),
      decoration: _cardDecoration(),
      child: const Column(
        children: [
          CircularProgressIndicator(),
          SizedBox(height: 12),
          Text(
            'Etkinlik detayı yükleniyor...',
            style: TextStyle(
              color: AppColors.grayText,
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildEmptyCard() {
    return Container(
      width: double.infinity,
      padding: const EdgeInsets.all(24),
      decoration: _cardDecoration(),
      child: const Column(
        children: [
          Icon(
            Icons.error_outline,
            size: 48,
            color: AppColors.grayText,
          ),
          SizedBox(height: 10),
          Text(
            'Etkinlik bilgisi bulunamadı.',
            textAlign: TextAlign.center,
            style: TextStyle(
              fontSize: 16,
              color: AppColors.darkText,
              fontWeight: FontWeight.bold,
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildContent(EventModel event) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        _buildPoster(event),
        const SizedBox(height: 14),
        _buildInfoCard(event),
      ],
    );
  }

  Widget _buildPoster(EventModel event) {
    final posterUrl = _posterUrl(event);

    return ClipRRect(
      borderRadius: BorderRadius.circular(18),
      child: SizedBox(
        width: double.infinity,
        height: 250,
        child: posterUrl.isEmpty
            ? _buildPosterPlaceholder()
            : Image.network(
          posterUrl,
          fit: BoxFit.cover,
          errorBuilder: (context, error, stackTrace) {
            return _buildPosterPlaceholder();
          },
          loadingBuilder: (context, child, loadingProgress) {
            if (loadingProgress == null) {
              return child;
            }

            return const Center(
              child: CircularProgressIndicator(),
            );
          },
        ),
      ),
    );
  }

  Widget _buildPosterPlaceholder() {
    return Container(
      color: const Color(0xFFE2E8F0),
      child: const Center(
        child: Icon(
          Icons.image_outlined,
          size: 56,
          color: AppColors.grayText,
        ),
      ),
    );
  }

  Widget _buildInfoCard(EventModel event) {
    final cityName = event.city?.name ?? event.cityName ?? '-';
    final districtName = event.district?.name ?? event.districtName ?? '-';
    final venueName = event.venue?.name ?? '-';
    final venueAddress = event.venue?.address ?? '-';
    final priceText = '${event.basePrice?.toInt() ?? 0} TL';
    final quotaText = 'Kalan: ${event.remainingQuota ?? 0}';

    return Container(
      width: double.infinity,
      padding: const EdgeInsets.all(16),
      decoration: _cardDecoration(),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            event.title,
            style: const TextStyle(
              fontSize: 25,
              fontWeight: FontWeight.bold,
              color: AppColors.darkText,
            ),
          ),
          const SizedBox(height: 10),
          Text(
            event.description ?? 'Açıklama bulunmuyor.',
            style: const TextStyle(
              fontSize: 15,
              color: AppColors.grayText,
              height: 1.35,
            ),
          ),
          const SizedBox(height: 16),
          const Divider(),
          const SizedBox(height: 8),
          _buildDetailLine(
            title: 'Tarih',
            value: event.eventDate ?? '-',
          ),
          _buildDetailLine(
            title: 'Konum',
            value: '$cityName / $districtName',
          ),
          _buildDetailLine(
            title: 'Sahne',
            value: venueName,
          ),
          _buildDetailLine(
            title: 'Adres',
            value: venueAddress,
          ),
          const SizedBox(height: 12),
          const Divider(),
          const SizedBox(height: 12),
          Row(
            children: [
              Text(
                priceText,
                style: const TextStyle(
                  fontSize: 24,
                  fontWeight: FontWeight.bold,
                  color: AppColors.green,
                ),
              ),
              const Spacer(),
              Container(
                padding: const EdgeInsets.symmetric(
                  horizontal: 12,
                  vertical: 8,
                ),
                decoration: BoxDecoration(
                  color: AppColors.blue.withOpacity(0.10),
                  borderRadius: BorderRadius.circular(12),
                ),
                child: Text(
                  quotaText,
                  style: const TextStyle(
                    fontSize: 14,
                    fontWeight: FontWeight.bold,
                    color: AppColors.blue,
                  ),
                ),
              ),
            ],
          ),
          const SizedBox(height: 18),
          AppButton(
            text: _buyButtonText(event),
            backgroundColor: _buyButtonColor(event),
            isLoading: _isBuying,
            onPressed: _canBuy(event) ? _buyTicket : null,
          ),
        ],
      ),
    );
  }

  Widget _buildDetailLine({
    required String title,
    required String value,
  }) {
    return Padding(
      padding: const EdgeInsets.only(bottom: 11),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            title,
            style: const TextStyle(
              fontSize: 13,
              color: AppColors.grayText,
            ),
          ),
          const SizedBox(height: 3),
          Text(
            value,
            style: const TextStyle(
              fontSize: 15,
              color: AppColors.darkText,
              fontWeight: FontWeight.bold,
            ),
          ),
        ],
      ),
    );
  }

  BoxDecoration _cardDecoration() {
    return BoxDecoration(
      color: AppColors.cardBackground,
      borderRadius: BorderRadius.circular(18),
      boxShadow: [
        BoxShadow(
          color: Colors.black.withOpacity(0.08),
          blurRadius: 12,
          offset: const Offset(0, 5),
        ),
      ],
    );
  }
}